[FIX] point_of_sale: run POS from another company

adwid opened the pull request:
In a multicompany configuration, when starting a POS of another company
than the current one, the loading will freeze.

To reproduce the error:
1. Create 2 companies C1 and C2
2. Make sure C1 has taxes
3. Create one POS per company: POS_C1 and POS_C2
4. Go on Point of Sale dashboard
5. On top right (the companies list):
	Check the two companies (so that you can see POS_C1 and POS_C2)
	If not already done, select C1
6. Start a new session with POS_C2

=> The loading will be blocked on "account.tax" step. While loading
POS_C2, the system actually tries to load C1 taxes but does not have the
corresponding access rights.
The issue was supposed to be fixed thanks to 29db04e, but this line https://github.com/odoo/odoo/blob/2c01da189a4f79cd72372442562bf7187ebda34c/addons/web/static/src/js/chrome/abstract_web_client.js#L132 overrides the `allow_companies_ids` defined in RPC response https://github.com/odoo/odoo/blob/1d1281bcc5717b7319a489d042a49cf82119a592/addons/point_of_sale/controllers/main.py#L51

A specific key (`pos_session_company_ids`) is now defined to circumvent
this behavior.

OPW-2388210

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/62619).

adwid commented:
(Correction of #62548 )

adwid commented:
@nim-odoo

nim-odoo reviewed (approved):
I'd add to the commit message:

This use case was supposed to be fixed with https://github.com/odoo/odoo/commit/29db04e8aff841daafc9821871d9c24c8472994d, but the `allowed_company_ids` key is replaced in:

https://github.com/odoo/odoo/blob/2c01da189a4f79cd72372442562bf7187ebda34c/addons/web/static/src/js/chrome/abstract_web_client.js#L132

To circumvent this behavior, we use a specific key `pos_session_company_ids`.

nim-odoo commented:
Btw, not the most elegant solution, but one of the most reliable.

pimodoo commented:
Like @nim-odoo said, I'd modify the commit message to understand why it doesn't work.

I think you should also set the "allowed_company_ids" key in the session_info['user_context'] in the frontend with the key received by the "pos_session_company_ids" because the user_context is used in the RPC sent to the server

adwid commented:
@pimodoo On a previous commit, I replaced this https://github.com/odoo/odoo/blob/2c01da189a4f79cd72372442562bf7187ebda34c/addons/web/static/src/js/chrome/abstract_web_client.js#L132 with  `session.user_context.allowed_company_ids |= stateCompanyIDS;`
That way, if undefined, `allowed_company_ids` is set to the standard value.
But considering a discussion with @nim-odoo, this is a very (too) early change in the stack and could create problems elsewhere.

(I updated the commit message)

nim-odoo reviewed (approved):
robodoo r+
